### PR TITLE
Rewrite purge-cluster playbook

### PR DESCRIPTION
**Pull request by KGoreczny**
Rewritten purge-cluster playbook. 
Tested with Ceph Hammer and Infernalis on CentOs 7 and Ubuntu 14.04 with Ansible 2.0.1.0

### CODE REVIEW

File: purge-cluster.yml [hunk 1]
```diff
@@ -2,42 +2,193 @@
...
+- name: stop ceph cluster
+  hosts:
+    - mons
+    - osds
+    - mdss
+  become: yes
+
+  vars:
+# When set to true both groups of packages are purged.
+# This can couse problem with qemu-kvm
```
Comment (leseb) [inline purge-cluster.yml]: s/couse/cause

**Comment by leseb** [comment]
@KGoreczny thanks for this :) just a few comments :)

File: purge-cluster.yml [hunk 2]
```diff
@@ -2,42 +2,193 @@
...
+  - name: get osd data mount points
+    shell: ls /var/lib/ceph/osd
+    register: mounted_osd
+    changed_when: false
+    failed_when: mounted_osd.rc != 0 and mounted_osd.rc != 2
+    when:
+     osd_group_name in group_names
+
+  - name: umount osd data partition
+    shell: umount /var/lib/ceph/osd/{{ item }}
```
Comment (leseb) [inline purge-cluster.yml]: mount module instead?

**Comment by leseb** [comment]
I'm a little bit hesitant since this only supports ansible v2. (because of the package module).

File: purge-cluster.yml (hunk 2 above)
Comment (KGoreczny) [inline purge-cluster.yml]: Problem with mount module is that it operates on /etc/fstab, not current system state. So for me it is not useful in this case as all osds are mounted dynamically, without any change is fstab.

Comment (leseb) [inline purge-cluster.yml]: Correct we don't need to touch `/etc/fstab`.

File: purge-cluster.yml [hunk 3]
```diff
@@ -2,42 +2,215 @@
...
+      ansible_distribution != 'Ubuntu'
+
+  - name: purge ceph packages with apt
+    apt:
+      name: "{{ item }}"
+      state: absent
+    with_items:
+      - "{{ ceph_packages }}"
+    when:
+      ansible_distribution == 'Ubuntu'
```
Comment (leseb) [inline purge-cluster.yml]: `when: ansible_pkg_mgr == "apt"`?

File: purge-cluster.yml [hunk 4]
```diff
@@ -2,42 +2,215 @@
...
+     osd_group_name in group_names
+
+  - name: purge ceph packages with yum
+    yum:
+      name: "{{ item }}"
+      state: absent
+    with_items:
+      - "{{ ceph_packages }}"
+    when:
+      ansible_distribution != 'Ubuntu'
```
Comment (leseb) [inline purge-cluster.yml]: `when: ansible_pkg_mgr == "yum"`?

File: purge-cluster.yml [hunk 5]
```diff
@@ -2,42 +2,215 @@
...
+      ansible_distribution == 'Ubuntu'
+
+  - name: purge remaining ceph packages with yum
+    yum:
+      name: "{{ item }}"
+      state: absent
+    with_items:
+      - "{{ ceph_remaining_packages }}"
+    when:
+      ansible_distribution != 'Ubuntu' and
```
Comment (leseb) [inline purge-cluster.yml]: when: ansible_pkg_mgr == "yum"?

File: purge-cluster.yml [hunk 6]
```diff
@@ -2,42 +2,215 @@
...
+      purge_all_packages == true
+
+  - name: purge remaining ceph packages with apt
+    apt:
+      name: "{{ item }}"
+      state: absent
+    with_items:
+      - "{{ ceph_remaining_packages }}"
+    when:
+      ansible_distribution == 'Ubuntu' and
```
Comment (leseb) [inline purge-cluster.yml]: when: ansible_pkg_mgr == "apt"?

File: purge-cluster.yml [hunk 7]
```diff
@@ -2,42 +2,234 @@
...
+     path: /var/log/ceph
+     state: absent
+
+  - name: remove form SysV
+    shell: "update-rc.d -f ceph remove"
+    when:
+      ansible_distribution == 'Ubuntu'
+
+  - name: remove Upstart nad SysV files
+    shell: 'find /etc -name "*ceph*" -delete'
```
Comment (leseb) [inline purge-cluster.yml]: Why do you sometimes use simple or double quotes for the shell module?
Double quotes all the time are probably the way to go.

**Comment by leseb** [comment]
Thanks @KGoreczny for addressing the comments.

**Pull request merged**